crates/frontend/src/utils.rs

```diff
@@ -343,3 +349,66 @@ pub fn get_config_value(
         }
     }
 }
+
+/********* Request Utils **********/
+
+use once_cell::sync::Lazy;
+static HTTP_CLIENT: Lazy<reqwest::Client> = Lazy::new(|| reqwest::Client::new());
+
+pub async fn request<'a, T, R>(
+    url: String,
+    method: reqwest::Method,
+    body: Option<T>,
+    headers: &[(&str, &str)],
```

Comment:
It maybe better to take `Headers` type directly instead of constructing it, just a thought

Reply:
If you see, you have to use `HeaderName::from_str` and `HeaderValue::from_str` on `&str` for header name and value, respectively, these functions then return a `Result`, which has to be handled to extract out the parsed value. Its a repeated procedure which has to be done for every request, seemed better to abstract it out.

Now that I am thinking about this, may be I can make another helper fxn to just construct the `HeaderMap` which I can then pass to the `request`.

Reply:
done